lib/govuk_tech_docs/tech_docs_html_renderer.rb

```diff
@@ -30,5 +30,34 @@ def table(header, body)
         </table>
       </div>)
     end
+
+    def table_row(body)
+      # Post-processing the table_cell HTML to implement row headings.
+      #
+      # Doing this in table_row instead of table_cell is a hack.
+      #
+      # Ideally, we'd use the table_cell callback like:
+      #
+      # def table_cell(content, alignment, header)
+      #   if header
+      #     "<th>#{content}</th>"
+      #   elsif content.start_with? "# "
+      #     "<th scope="row">#{content.sub(/^# /, "")}</th>"
+      #   else
+      #     "<td>#{content}</td>"
+      #   end
+      # end
+      #
+      # Sadly, Redcarpet's table_cell callback doesn't allow you to distinguish
+      # table cells and table headings until https://github.com/vmg/redcarpet/commit/27dfb2a738a23aadd286ac9e7ecd61c4545d29de
+      # (which is not yet released). This means we can't use the table_cell callback
+      # without breaking column headers, so we're having to hack it in table_row.
+      #
+      # Despite the hackiness, this substitution should work for all cases
+      # where the table cell just includes simple text (with no other markup).
+
+      body_with_row_headers = body.gsub(/<td># ([^<]*)<\/td>/, "<th scope=\"row\">\\1</th>")
```

Comment:
Am I right in thinking the space between the `#` and the text is required for this pattern to match?

From testing, it looks like redcarpet will happily make something a heading without the space (e.g. `#Hello world` becomes an H1), so I'm wondering if it's worth being consistent on that here.

(Conversely, CommonMark [requires the space, but does acknowledge that many implementations don't](https://spec.commonmark.org/0.29/#atx-headings))

Reply:
Yeah, this pattern requires it. I wasn't sure whether it was needed in the spec or not.

🤔  🤔  Not sure whether to keep the space or not...